Dockerfile

```diff
@@ -5,21 +5,30 @@ FROM python:3.12
 RUN echo 'Package: nodejs\nPin: origin deb.nodesource.com\nPin-Priority: 600' > /etc/apt/preferences.d/nodesource \
     && curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
     && apt-get update \
-    && apt-get install -y nodejs
+    && apt-get install -y nodejs \
+    && rm -rf /var/lib/apt/lists/*
 
 # Install yarn
 RUN npm install -g yarn
 
 # Set working directory
 WORKDIR /app
 
-# Copy pyproject file to the working directory
-COPY pyproject.toml .
+# Install uv
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/
 
-# Install Python dependencies
-RUN pip install --no-cache-dir .
+# Set up uv environment for caching
+# See:
+# - https://docs.astral.sh/uv/guides/integration/docker/#caching
+ENV UV_LINK_MODE=copy
```

Comment:
These docs detail why this setup. TLDR: avoiding code changes causing cache busting of expensive deps layers.